### PR TITLE
feat: add node power types and events

### PR DESCRIPTION
**Pull request by renauter**
Related to:

- #88

### CODE REVIEW

**Comment by renauter** [comment]
Tested on local node
Node 1 and Node  2 belong to Farm 1

image

File: src/mappings/power.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import {
+    EventHandlerContext,
+    Store
+} from "@subsquid/substrate-processor";
+import { NodePower, PowerState, Power } from "../model";
+import { TfgridModulePowerTargetChangedEvent, TfgridModulePowerStateChangedEvent } from "../types/events";
+import assert from 'assert'
+
+export async function powerTargetChanged(ctx: EventHandlerContext) {
+    let powerTargetChangedEvent = new TfgridModulePowerTargetChangedEvent(ctx).asV123
```
Comment (DylanVerstraete) [inline src/mappings/power.ts:10]: destructure fields for easier access:

```
let { nodeId, farmId, powerTarget } = new TfgridModulePowerTargetChangedEvent(ctx).asV123
```

File: src/mappings/power.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        newNodePower.farmID = powerTargetChangedEvent.farmId
+        newNodePower.nodeID = powerTargetChangedEvent.nodeId
+        newNodePower.state = PowerState.Up
+        newNodePower.target = target
+        await ctx.store.save<NodePower>(newNodePower)
+    }
+}
+
+export async function powerStateChanged(ctx: EventHandlerContext) {
+    let powerStateChangedEvent = new TfgridModulePowerStateChangedEvent(ctx).asV123
```
Comment (DylanVerstraete) [inline src/mappings/power.ts:40]: destructure fields for easier access:

```
let { nodeId, farmId, powerTarget } = new TfgridModulePowerTargetChangedEvent(ctx).asV123
```

**Review by DylanVerstraete — CHANGES_REQUESTED** [review]
I think the nodepower should be part of the general node struct (just a json field) and not a seperate entity

**Comment by renauter** [comment]
> I think the nodepower should be part of the general node struct (just a json field) and not a seperate entity

You mean in the `schema.graphql` file I guess?

**Comment by DylanVerstraete** [comment]
> > I think the nodepower should be part of the general node struct (just a json field) and not a seperate entity
> 
> You mean in the `schema.graphql` file I guess?

Indeed

**Comment by DylanVerstraete** [comment]
@renauter can you also rebase your changes because I just merged the firesquid branch

**Comment by DylanVerstraete** [comment]
closing in favor of https://github.com/threefoldtech/tfchain_graphql/pull/92

**Pull request closed without merging**